### DISCUSSION TITLE
How do you add an image logo to the Title header?

### DISCUSSION THREAD

**Discussion by enjoysmath** [dotnet/maui#11884] [Q&A] [answered]
Right now my title header just consists of a text string on my Content Pages but I would like a logo image there.  How can it be done?

**Comment by mhrastegari** [comment, marked as answer]
You can use `TitleView` for that.
Here's a sample for you :

```
<ContentPage....
    <NavigationPage.TitleView>    
        <StackLayout HorizontalOptions="Center" VerticalOptions="Center"  Orientation="Horizontal">    
            <Image HorizontalOptions="Start" Source="sample.png" />
            <Label HorizontalOptions="Start" VerticalTextAlignment="Center" Text="MyPage" FontSize="Small"/>    
        </StackLayout>    
    </NavigationPage.TitleView>
    <Grid....
</ContentPage>
```

**Comment by enjoysmath** [comment]
@mhrastegari 

That doesn't work:
```
<?xml version="1.0" encoding="UTF-8" ?>
<ContentPage x:Class="CEDDashboard.View.SettingsDetailPage"
             xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:viewmodel="clr-namespace:CEDDashboard.ViewModel"
             xmlns:local="clr-namespace:CEDDashboard.View"
             x:DataType="viewmodel:SettingsDetailViewModel"
             Title="{Binding Device.Name}">

    <NavigationPage.TitleView>
        <StackLayout HorizontalOptions="Center" VerticalOptions="Center"  Orientation="Horizontal">
            <Image HorizontalOptions="Start" Source="app_logo.png" />
            <Label HorizontalOptions="Start" VerticalTextAlignment="Center" Text="Device Detail Settings" FontSize="Small"/>
        </StackLayout>
    </NavigationPage.TitleView>
    
    <ContentPage.Content>
        <VerticalStackLayout>
            <Image HorizontalOptions="Start" Source="app_logo.png" />
            <ScrollView BackgroundColor="{AppThemeBinding Light={StaticResource LightBackground}, Dark={StaticResource DarkBackground}}">
                <Grid RowDefinitions="Auto,Auto,*">
                    <BoxView

```
The only image that is displayed is the one inside my content.  The NavigationPage.TitleView trick doesn't show any image.

**Reply by mhrastegari** [reply]
Did you changed
`MainPage = new MainPage();
`
to
`MainPage = new NavigationPage(new MainPage());
`
in `App.xaml.cs` file?

**Comment by enjoysmath** [comment]
Mohammad,

This is what I currently have:











*namespace CEDDashboard;public partial class App : Application{ public
App() { InitializeComponent(); MainPage = new AppShell(); }}*

Thanks!

Daniel Donnelly, Jr.

**Comment by enjoysmath** [comment]
I tried wrapping new AppShell() in a new NavigationPage(), the result is a
crash every interaction with my app, without error shown

**Reply by mhrastegari** [reply]
If you're using the `Shell` all you need to do is change that `<NavigationPage.TitleView>` to `<Shell.TitleView>`.

You can read about Shell `TitleView` [here](https://learn.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/shell/pages#display-views-in-the-navigation-bar).

**Reply by enjoysmath** [reply]
Thanks!  That worked.

**Reply by mhrastegari** [reply]
Happy to help, If it worked can you mark my comment as the accepted answer?

**Reply by enjoysmath** [reply]
It's not letting me mark it as answer
